src/dbnode/storage/index/allocator.go

```diff
@@ -23,15 +23,15 @@ package index
 import (
 	"github.com/m3db/m3/src/dbnode/storage/bootstrap/result"
 	"github.com/m3db/m3/src/m3ninx/index/segment"
-	"github.com/m3db/m3/src/m3ninx/index/segment/mem"
+	"github.com/m3db/m3/src/m3ninx/index/segment/builder"
 )
 
-// NewBootstrapResultMutableSegmentAllocator returns a default mutable segment
+// NewBootstrapResultDocumentsBuilderAllocator returns a default mutable segment
 // allocator for a bootstrap result index block given index options.
-func NewBootstrapResultMutableSegmentAllocator(
+func NewBootstrapResultDocumentsBuilderAllocator(
```

Comment:
nit: Should we rename this `NewBootstrapDocumentsBuilderAllocator(...)` to match the new type?